tests/common.rs

```diff
@@ -9,20 +9,49 @@ use std::future::Future;
 use std::net::{IpAddr, Ipv4Addr, SocketAddr};
 use std::path::PathBuf;
 use warp::Filter;
+use fantoccini::ClientBuilder;
+use fantoccini::Client;
 
 #[cfg(feature = "rustls-tls")]
-pub type Client = fantoccini::RustlsClient;
-#[cfg(all(feature = "openssl-tls", not(feature = "rustls-tls")))]
-pub type Client = fantoccini::OpenSslClient;
+pub async fn rustls_type(s: &str) -> Result<Client<hyper_rustls::HttpsConnector<hyper::client::HttpConnector>>, error::NewSessionError> {
+    match s {
+        "firefox" => {
+            let mut caps = serde_json::map::Map::new();
+            let opts = serde_json::json!({ "args": ["--headless"] });
+            caps.insert("moz:firefoxOptions".to_string(), opts.clone());
+            ClientBuilder::rustls().capabilities(caps).connect("http://localhost:4444").await
+        }
+        "chrome" => {
+            let mut caps = serde_json::map::Map::new();
+            let opts = serde_json::json!({
+                "args": ["--headless", "--disable-gpu", "--no-sandbox", "--disable-dev-shm-usage"],
+                "binary":
+                    if std::path::Path::new("/usr/bin/chromium-browser").exists() {
+                        // on Ubuntu, it's called chromium-browser
+                        "/usr/bin/chromium-browser"
+                    } else if std::path::Path::new("/Applications/Google Chrome.app/Contents/MacOS/Google Chrome").exists() {
+                        // macOS
+                        "/Applications/Google Chrome.app/Contents/MacOS/Google Chrome"
+                    } else {
+                        // elsewhere, it's just called chromium
+                        "/usr/bin/chromium"
+                    }
+            });
+            caps.insert("goog:chromeOptions".to_string(), opts.clone());
```

Comment:
Let's try to put the (shared) code for constructing the capabilities into its own function which the two TLS configurations can then call out to.